### PR TITLE
Combine frontend navigation Page list block tests into one test to speed up e2e tests

### PR DESCRIPTION
**Pull request by jeryj**
## What?

Combines several similar tests with repeated steps for Frontend Navigation menu tests into one test.

## Why?

The setup for each test was taking awhile, and there were many repeated steps in each test. Combining them follows best practices of [writing fewer but longer tests](https://playwright.dev/docs/best-practices#write-fewer-tests-but-longer-tests) and saves quite a bit of time. 

This branch's tests: 25 seconds
Same tests on trunk: 2 minutes

## How?

Take the previous tests for the frontend navigation and combine them into one where it shared interactions and code.

## Testing Instructions
`npm run test:e2e:playwright -- editor/blocks/navigation.spec.js -g "Navigation block - Frontend interactivity"` will run the test touched by this code

### CODE REVIEW

**Review by kevin940726 — APPROVED** [review]
Nice improvements! Thanks!

File: test/e2e/artifacts/storage-states/admin.json [hunk 1]
```diff
@@ -0,0 +1 @@
...H9s4ikPkhkS07FzxnWES5FY2SsnwL7v%7C45404d74460259bc9148f2357f2180af488d65921b10ed5981fff860afa5c8ca","domain":"localhost","path":"/wp-content/plugins","expires":-1,"httpOnly":true,"secure":false,"sameSite":"Lax"},{"name":"wordpress_23778236db82f19306f247e20a353a99","value":"admin%7C1684446357%7Cdx01QADx42SHH9s4ikPkhkS07FzxnWES5FY2SsnwL7v%7C45404d74460259bc9148f2357f2180af488d65921b10ed5981fff860afa5c8ca","domain":"localhost","path":"/wp-admin","expires":-1,"httpOnly":true,"secure":false,"sameSite":"Lax"},{"name":"wordpress_logged_in_23778236db82f19306f247e20a353a99","value":"admin%7C1684446357%7Cdx01QADx42SHH9s4ikPkhkS07FzxnWES5FY2SsnwL7v%7C8ace4a8f867bc4e587d5264662296f90fcd133710cd0dd3386a92801816bd5d1","domain":"localhost","path":"/","expires":-1,"httpOnly":true,"secure":false,"sameSite":"Lax"},{"name":"wp-settings-1","value":"editor%3Dtinymce","domain":"localhost","path":"/","expires":1715809558.14,"httpOnly":false,"secure":false,"sameSite":"Lax"},{"name":"wp-settings-time-1","value":"1684273558","domain":"localhost","path":"/","expires":1715809558.14,"httpOnly":false,"secure":false,"sameSite":"Lax"}],"nonce":"fe590a7aae","rootURL":"http://localhost:8889/index.php?rest_route=/"}
```
Comment (kevin940726) [inline test/e2e/artifacts/storage-states/admin.json:1]: I don't think this was supposed to get committed? Is this an oversight?

Comment (jeryj) [reply test/e2e/artifacts/storage-states/admin.json:1]: Oof. Definitely not intended to be committed. I'll remove it.

Comment (jeryj) [reply test/e2e/artifacts/storage-states/admin.json:1]: https://github.com/WordPress/gutenberg/pull/50795

**Pull request merged**